src/selectors/entities/channel_categories.test.js

```diff
@@ -191,6 +200,418 @@ describe('makeFilterChannelsByType', () => {
     });
 });
 
+describe('makeFilterAutoclosedDMs', () => {
+    const currentUser = {id: 'currentUser'};
+
+    const baseState = {
+        entities: {
+            channels: {
+                currentChannelId: 'channel1',
+                myMembers: {
+                    channel1: {},
+                },
+            },
+            general: {
+                config: {
+                    CloseUnusedDirectMessages: 'true',
+                },
+            },
+            posts: {
+                posts: {},
+                postsInChannel: {
+                    channel1: [],
+                },
+            },
+            preferences: {
+                myPreferences: {
+                    [getPreferenceKey(Preferences.CATEGORY_SIDEBAR_SETTINGS, Preferences.CHANNEL_SIDEBAR_AUTOCLOSE_DMS)]: {value: Preferences.AUTOCLOSE_DMS_ENABLED},
+                },
+            },
+            users: {
+                currentUserId: currentUser.id,
+                profiles: {
+                    currentUser,
+                },
+            },
+        },
+    };
+
+    const now = Date.now();
+    const cutoff = now - 7 * 24 * 60 * 60 * 1000;
+
+    function isChannelVisiblePrecondition(state, channel) {
+        return isGroupOrDirectChannelVisible(
+            channel,
+            getMyChannelMemberships(state),
+            getConfig(state),
+            getMyPreferences(state),
+            getCurrentUserId(state),
+            state.entities.users.profiles,
+            getLastPostPerChannel(state),
+            getCurrentChannelId(state),
+            now,
+        );
+    }
+
+    test('should hide an inactive GM channel', () => {
+        const filterAutoclosedDMs = Selectors.makeFilterAutoclosedDMs(() => cutoff);
+
+        const channel1 = {id: 'channel1', type: General.GM_CHANNEL};
+
+        const state = mergeObjects(baseState, {
```

Comment:
Also, I'm trying out a new method of merging state objects to see if that makes these test cases more readable when they slightly modify the existing state. Let me know if you have any thoughts on how this looks. You can see the difference between this version and the full state here: https://github.com/mattermost/mattermost-redux/pull/1075/commits/dfda2ac8461fbf0e368f46de49bc7eacacbc7ef4

Reply:
I think if anything we just need to be louder about the fact that the method exists so that more people use it.
Also document any limitations if they exist

Reply:
Nice 👍